kubeflow/trainer/api/trainer_client.py

```
@@ -12,576 +12,208 @@
-        self.namespace = namespace
+        """
+        # initialize training backend
+        if not backend_config:
+            backend_config = KubernetesBackendConfig()
+
+        if isinstance(backend_config, KubernetesBackendConfig):
+            self.backend = KubernetesBackend(backend_config)
+        elif isinstance(backend_config, LocalProcessBackendConfig):
+            self.backend = LocalProcessBackend(backend_config)
+        else:
+            raise ValueError(f"Invalid backend config '{backend_config}'")
```

Ruff (0.13.1)

Prefer `TypeError` exception for invalid type

(TRY004)